src/test/java/de/hs_mannheim/informatik/ct/end_to_end/BasicRoomControllerTest.java

```diff
@@ -0,0 +1,94 @@
+package de.hs_mannheim.informatik.ct.end_to_end;
+
+/*
+ * Corona Tracking Tool der Hochschule Mannheim
+ * Copyright (C) 2021 Hochschule Mannheim
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import de.hs_mannheim.informatik.ct.model.Room;
+import de.hs_mannheim.informatik.ct.persistence.InvalidEmailException;
+import de.hs_mannheim.informatik.ct.persistence.InvalidExternalUserdataException;
+import de.hs_mannheim.informatik.ct.persistence.services.RoomService;
+import de.hs_mannheim.informatik.ct.persistence.services.RoomVisitService;
+import de.hs_mannheim.informatik.ct.persistence.services.VisitorService;
+import org.junit.jupiter.api.BeforeEach;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.web.servlet.MockMvc;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.ANY)
+abstract public class BasicRoomControllerTest {
+    @TestConfiguration
+    static class RoomControllerTestConfig {
+        @Bean
+        public RoomService service() {
+            return new RoomService();
+        }
+    }
+
+    @Autowired
+    protected RoomService roomService;
+
+    @Autowired
+    protected RoomVisitService roomVisitService;
+
+    @Autowired
+    protected VisitorService visitorService;
+
+    @Autowired
+    protected MockMvc mockMvc;
+
+    protected final String TEST_ROOM_NAME = "123";
+    protected String TEST_ROOM_PIN;
+    protected final String TEST_ROOM_PIN_INVALID = "";
+    protected final String TEST_USER_EMAIL = "1233920@stud.hs-mannheim.de";
```

Comment:
all attributes and util methods are now in the base class. In the subclasses are only the tests